crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -52,234 +105,210 @@ pub enum SceneSpawnError {
-                    .get(&scene_handle)
-                    .ok_or_else(|| SceneSpawnError::NonExistentRealScene {
-                        handle: scene_handle.clone(),
-                    })?;
-
-            for archetype in scene.world.archetypes().iter() {
-                for scene_entity in archetype.entities() {
-                    let entity = *instance_info
-                        .entity_map
-                        .entry(*scene_entity)
-                        .or_insert_with(|| world.spawn().id());
-                    for component_id in archetype.components() {
-                        let component_info = scene
-                            .world
-                            .components()
-                            .get_info(component_id)
-                            .expect("component_ids in archetypes should have ComponentInfo");
-
-                        let reflect_component = type_registry
-                            .get(component_info.type_id().unwrap())
-                            .ok_or_else(|| SceneSpawnError::UnregisteredType {
-                                type_name: component_info.name().to_string(),
-                            })
-                            .and_then(|registration| {
-                                registration.data::<ReflectComponent>().ok_or_else(|| {
-                                    SceneSpawnError::UnregisteredComponent {
-                                        type_name: component_info.name().to_string(),
-                                    }
-                                })
-                            })?;
-                        reflect_component.copy_component(
-                            &scene.world,
-                            world,
-                            *scene_entity,
-                            entity,
-                        );
-                    }
-                }
-            }
-            for registration in type_registry.iter() {
-                if let Some(map_entities_reflect) = registration.data::<ReflectMapEntities>() {
-                    map_entities_reflect
-                        .map_entities(world, &instance_info.entity_map)
-                        .unwrap();
-                }
-            }
-            self.spawned_instances.insert(instance_id, instance_info);
-            let spawned = self
-                .spawned_scenes
-                .entry(scene_handle)
-                .or_insert_with(Vec::new);
-            spawned.push(instance_id);
-            Ok(instance_id)
-        })
+        let mut entity_map = EntityMap::default();
+        scene_handle.write_to_world(world, &mut entity_map)?;
+
+        self.instances_info
+            .insert(instance_id, InstanceInfo { entity_map });
+        let spawned = self.instances.entry(scene_handle).or_insert_with(Vec::new);
+        spawned.push(instance_id);
+        Ok(instance_id)
     }
 
-    pub fn update_spawned_scenes(
+    fn update_spawned_scenes(
```

Comment:
I made this method private simply out of laziness. Note that it doesn't really make sense to expose this publicly either.

Reply:
I made it public again, I needed to export `SceneHandle`.